src/openms/source/ANALYSIS/ID/SimpleSearchEngineAlgorithm.cpp

```diff
@@ -340,6 +340,14 @@ void SimpleSearchEngineAlgorithm::postProcessHits_(const PeakMap& exp,
       }
     }
 
+#ifdef _OPENMP
+    // we need to sort the peptide_ids by scan_index in order to have the same output in the idXML-file
+    std::sort(peptide_ids.begin(), peptide_ids.end(), [](const PeptideIdentification& a, const PeptideIdentification& b)
```

Comment:
ouch. this looks expensive?

you can omit the sorting, if #threads == 1.
The ifdef usually does not help a lot, since most of the time, OpenMS is compiled with openmp enabled.

Reply:
sorting should still be fine. I would say 95% of all time are spend in spectrum generation and comparison

Reply:
So, should it be `if (omp_get_num_threads() > 1) sort();` ? Can I access the number of threads outside of a parallel region?

Reply:
no, should be `omp_get_max_threads()`
see  https://blogs.oracle.com/d/openmp-getting-the-maximum-number-of-threads

Reply:
Okay, but should I leave the #ifdef _OPENMP? omp_get_num_threads() is not defined for certain systems, right?